store/tikv/2pc.go

```diff
@@ -616,6 +617,9 @@ func (c *twoPhaseCommitter) pessimisticLockSingleBatch(bo *Backoffer, batch batc
 				}
 				return errors.Trace(conditionPair.Err())
 			}
+			if deadlock := keyErr.Deadlock; deadlock != nil {
+				return errors.New("deadlock")
```

Comment:
It's better to use the same error as MySQL.

Reply:
It's replaced to MySQL deadlock error here:
https://github.com/pingcap/tidb/blob/master/session/tidb.go#L166

Reply:
This is a bad design while matching string. Could you refactor it?